src/chain/store/chain_store.rs

```diff
@@ -200,8 +193,14 @@ where
 
     /// Returns the currently tracked heaviest tipset.
     pub fn heaviest_tipset(&self) -> Arc<Tipset> {
-        self.tipset_from_keys(self.file_backed_heaviest_tipset_keys.lock().inner())
-            .expect("Failed to load heaviest tipset")
+        self.tipset_from_keys(
+            &self
+                .settings
+                .read_obj::<TipsetKeys>(HEAD_KEY)
+                .expect("failed to load heaviest tipset")
+                .expect("failed to load heaviest tipset"),
+        )
+        .expect("failed to load heaviest tipset")
```

Comment:
Hmm, is there a way to reduce the number of duplicate error messages? Maybe we need a `require_obj` which returns `Err` instead of `None` when a key is missing.

Reply:
Splendid idea!